src/fhda/tortoise.py

```diff
@@ -0,0 +1,260 @@
+import os
+import uuid
+import asyncio
+import copy
+from typing import Any, Callable, Optional
+from os import PathLike
+import time
+import json
+from pydantic import BaseModel, Field
+from . import prompts
+
+from futurehouse_client import FutureHouseClient
+from futurehouse_client.models import TaskRequest, RuntimeConfig
+from futurehouse_client.models.app import AuthType
+
+
+class StepConfig(BaseModel):
+    """Configuration for a step in the pipeline."""
+
```

Comment:
What doesn't make sense to me is `Step` has extensive configuration as many `Field`s, then we also have a separate `StepConfig`. It would be good to either:
- Rename `StepConfig` to be clearer
- Consolidate `StepConfig` into `Step`